vision/why-compile-time-evaluation.md

```diff
@@ -0,0 +1,44 @@
+# Why compile time evaluation?
+
+This section is, in many ways, the most important. It aims to clarify the reasoning about any future decisions about const generics and const evaluation.
+
+## Why do we want to evaluate things at compile time?
```

Comment:
one point not touched on here is correctness. People want to check that their datastructures are used correctly, ideally with a compiler error instead of a panic.